broker/data-access-layer/eventmesh/src/ApiServerClient.js

```diff
@@ -597,30 +630,57 @@ class ApiServerClient {
     assert.ok(opts.resourceType, 'Property \'resourceType\' is required to get resource');
     assert.ok(opts.resourceId, 'Property \'resourceId\' is required to get resource');
     const namespaceId = opts.namespaceId ? opts.namespaceId : this.getNamespaceId(opts.resourceId);
-    return Promise.try(() => apiserver.apis[opts.resourceGroup][CONST.APISERVER.API_VERSION]
-      .namespaces(namespaceId)[opts.resourceType](opts.resourceId).get())
-      .then(resource => {
-        _.forEach(resource.body.spec, (val, key) => {
+
+    const crdJson = this.getCrdJson(opts.resourceGroup, opts.resourceType);
+    const group = crdJson ? crdJson.spec.group : opts.resourceGroup;
+    const version = this.getCrdVersion(crdJson);
+    const plural = crdJson ? crdJson.spec.names.plural : _.lowerCase(opts.resourceType);
+
+    const client = this._getApiClient(group, version);
+    return Promise.try(() => client.getNamespacedCustomObject(group, version, namespaceId, plural, opts.resourceId))
+      .then(response => {
+        _.forEach(response.body.spec, (val, key) => {
           try {
-            resource.body.spec[key] = JSON.parse(val);
+            response.body.spec[key] = JSON.parse(val);
           } catch (err) {
-            resource.body.spec[key] = val;
+            response.body.spec[key] = val;
           }
         });
-        _.forEach(resource.body.status, (val, key) => {
+        _.forEach(response.body.status, (val, key) => {
           try {
-            resource.body.status[key] = JSON.parse(val);
+            response.body.status[key] = JSON.parse(val);
           } catch (err) {
-            resource.body.status[key] = val;
+            response.body.status[key] = val;
           }
         });
-        return resource.body;
+        return response.body;
       })
       .catch(err => {
         return convertToHttpErrorAndThrow(err);
       });
   }
 
+  /**
+   * @description Get a Resource which is cluster scoped in Apiserver with the opts
+   * @param {string} opts.resourceGroup - Group of resource the resource. eg: osb.servicefabrik.io
+   * @param {string} opts.resourceVersion - optional; api version of resource eg: v1alpha1
+   * @param {string} opts.resourceType - Kind of the resource. eg: sfserviceinstances (plural)
+   * @param {string} opts.resourceId - Id of resource
+   */
+  getClusterResource(opts) {
```

Comment:
Can we rename it as `getClusterScopedResource`

Reply:
Is an unused function. Was added when i was learning the sdk. Will remove the function